File: abc/abc161/abc161_d.cs
// https://atcoder.jp/contests/abc161/submissions/11612594

using System;
using System.Collections.Generic;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var K = long.Parse(Console.ReadLine());

            var queue = new Queue<long>();
            for (var i = 1; i <= 9; i++)
            {
                queue.Enqueue(i);
            }

            var n = 0L;
            for (var i = 1; i <= K; i++)
            {
                n = queue.Dequeue();
                if (n % 10 != 0)
                {
                    queue.Enqueue(n * 10 + n % 10 - 1);
                }
                queue.Enqueue(n * 10 + n % 10);
                if (n % 10 != 9)
                {
                    queue.Enqueue(n * 10 + n % 10 + 1);
                }
            }
            Console.WriteLine(n);
        }
    }
}

File: abc/abc161/abc161_d.fs
// https://atcoder.jp/contests/abc161/submissions/11747064

open System.Collections.Generic

let K = stdin.ReadLine() |> int

let queue = new Queue<int64>()

[ 1L..9L ] |> List.iter queue.Enqueue

let mutable n = 0L

let rec solver k =
    n <- queue.Dequeue()
    match n % 10L with
    | 0L ->
        [ n * 10L + n % 10L
          n * 10L + n % 10L + 1L ]
    | 9L ->
        [ n * 10L + n % 10L - 1L
          n * 10L + n % 10L ]
    | _ ->
        [ n * 10L + n % 10L - 1L
          n * 10L + n % 10L
          n * 10L + n % 10L + 1L ]
    |> List.iter queue.Enqueue
    if k = K - 1 then ()
    else solver (k + 1)

solver 0
n |> printfn "%d"
